packages/twenty-front/src/modules/favorites/components/FavoriteFoldersSelectEffects.tsx

```diff
@@ -0,0 +1,75 @@
+import { useEffect } from 'react';
+import { useRecoilCallback, useRecoilState, useSetRecoilState } from 'recoil';
+import { isDefined } from 'twenty-ui';
+import { isDeeplyEqual } from '~/utils/isDeeplyEqual';
+
+import { useFavoriteFolders } from '@/favorites/hooks/useFavoriteFolders';
+import { useFavorites } from '@/favorites/hooks/useFavorites';
+import { FavoriteFolder } from '@/favorites/types/FavoriteFolder';
+import { ObjectRecord } from '@/object-record/types/ObjectRecord';
+import { useFavoriteFoldersScopedStates } from '../hooks/useFavoriteFoldersScopedStates';
+
+type FavoriteFoldersMultiSelectEffectProps = {
+  record?: ObjectRecord;
+};
+
+export const FavoriteFoldersMultiSelectEffect = ({
+  record,
+}: FavoriteFoldersMultiSelectEffectProps) => {
+  const {
+    favoriteFoldersIdsMultiSelectState,
+    favoriteFolderMultiSelectFamilyState,
+    favoriteFoldersMultiSelectCheckedState,
+  } = useFavoriteFoldersScopedStates();
+
+  const { favoriteFolder } = useFavoriteFolders();
+  const { favorites } = useFavorites();
+  const setCheckedState = useSetRecoilState(
+    favoriteFoldersMultiSelectCheckedState,
+  );
+
+  const [favoriteFoldersIdsMultiSelect, setFavoriteFoldersIdsMultiSelect] =
+    useRecoilState(favoriteFoldersIdsMultiSelectState);
+
+  const updateFolders = useRecoilCallback(
+    ({ snapshot, set }) =>
+      (folders: FavoriteFolder[]) => {
+        folders.forEach((folder) => {
+          const currentFolder = snapshot
+            .getLoadable(favoriteFolderMultiSelectFamilyState(folder.id))
+            .getValue();
+
+          if (!isDeeplyEqual(folder, currentFolder)) {
+            set(favoriteFolderMultiSelectFamilyState(folder.id), folder);
+          }
+        });
+      },
+    [favoriteFolderMultiSelectFamilyState],
+  );
+
+  useEffect(() => {
```

Comment:
do you really need those useEffects? Can't you instead trigger the right function directly from the event that should trigger it

Reply:
will, get back to you! But this is how other multi selects are written in the project -- I followed the same pattern.
I knew you would complain seeing that many useEffects in a single component haha